### Midterm Python
### Daniel Kessler
### 33073652@fsv.cuni.cz

In [2]:

import matplotlib.pyplot as plt

import requests
import json

import os
import pandas as pd
import numpy as np

# TASK 1

loading the data and creating column with tha name of each company

In [73]:
df_ame = pd.read_csv ('AME.csv')
df_aph = pd.read_csv ('APH.csv')
df_elv = pd.read_csv ('ELV.csv')
df_hd = pd.read_csv ('HD.csv')
df_ame["Company"]="AME"
df_aph["Company"]="APH"
df_elv["Company"]="ELV"
df_hd["Company"]="HD"
print(df_hd)

            Date        High         Low        Open       Close      Volume  \
0     2017-11-13  166.639999  164.250000  164.639999  165.350006   7189300.0   
1     2017-11-14  168.139999  163.149994  163.619995  168.059998  10284000.0   
2     2017-11-15  167.440002  165.350006  167.440002  165.470001   5534500.0   
3     2017-11-16  167.759995  166.059998  166.679993  167.500000   4587900.0   
4     2017-11-17  168.429993  166.059998  167.000000  167.740005   4530100.0   
...          ...         ...         ...         ...         ...         ...   
1254  2022-11-07  291.700012  282.839996  286.500000  290.269989   3260200.0   
1255  2022-11-08  293.339996  287.630005  292.429993  291.140015   3371800.0   
1256  2022-11-09  292.500000  286.260010  289.000000  286.750000   3071400.0   
1257  2022-11-10  314.739990  297.079987  297.739990  311.700012   7794000.0   
1258  2022-11-11  317.679993  310.239990  312.000000  314.940002   4940400.0   

       Adj Close Company  
0     149.87

Creating pandas dataframe

In [74]:
df=pd.DataFrame(data=pd.concat([df_ame,df_aph,df_elv,df_hd]))
df.set_index("Date",inplace=True)
print(df)

                  High         Low        Open       Close     Volume  \
Date                                                                    
2017-11-13   68.989998   68.029999   68.129997   68.889999  1953500.0   
2017-11-14   69.500000   68.629997   68.629997   69.489998  1927400.0   
2017-11-15   69.440002   68.639999   69.010002   68.820000  1813800.0   
2017-11-16   70.430000   69.000000   69.230003   70.120003  1985100.0   
2017-11-17   70.430000   69.690002   69.949997   70.260002  2939400.0   
...                ...         ...         ...         ...        ...   
2022-11-07  291.700012  282.839996  286.500000  290.269989  3260200.0   
2022-11-08  293.339996  287.630005  292.429993  291.140015  3371800.0   
2022-11-09  292.500000  286.260010  289.000000  286.750000  3071400.0   
2022-11-10  314.739990  297.079987  297.739990  311.700012  7794000.0   
2022-11-11  317.679993  310.239990  312.000000  314.940002  4940400.0   

             Adj Close Company  
Date             

#### 1.1 I have the company in one name column, nvm

In [66]:
un=df.Company.unique()
un

array(['AME', 'APH', 'ELV', 'HD'], dtype=object)

In [68]:
un=df.Company.unique()
un
ame=df_ame["Adj Close"]==df_ame["Close"]
ame.unique()

array([False,  True])

In [69]:
aph=df_ame["Adj Close"]==df_aph["Close"]
aph.unique()

array([False])

In [75]:
elv=df_elv["Adj Close"]==df_elv["Close"]
elv.unique()

array([ True])

In [76]:
hd=df_hd["Adj Close"]==df_hd["Close"]
hd.unique()

array([False,  True])

Only for the company ELV, it is the same.

#### 1.2

In [77]:
df.groupby("Company").max()

High         Low        Open       Close      Volume  \
Company                                                               
AME      148.070007  146.369995  147.729996  147.410004   5586200.0   
APH       88.449997   86.940002   88.080002   87.940002   9182600.0   
ELV      549.520020  538.900024  545.000000  546.770020   7352000.0   
HD       420.609985  413.019989  417.000000  416.179993  19752800.0   

          Adj Close  
Company              
AME      146.637558  
APH       87.189758  
ELV      546.770020  
HD       416.179993

For AME, the highest Adj Close is 146.637558 <br />
For APH, the highest Adj Close is 87.189758 <br />
For ELV, the highest Adj Close is 546.770020<br />
For HD, the highest Adj Close is 416.179993<br />

In [78]:
df.groupby("Company").min()

High         Low        Open       Close     Volume   Adj Close
Company                                                                       
AME       62.900002   54.820000   60.549999   58.209999   180500.0   57.247528
APH       34.224998   31.525000   32.779999   34.169998   331400.0   33.339298
ELV      189.580002  171.029999  183.979996  174.679993   224500.0  174.679993
HD       161.399994  140.630005  150.059998  152.149994  1093900.0  146.575485

For AME, the lowest Adj Close is 57.247528 <br />
For APH, the lowest Adj Close is 33.339298 <br />
For ELV, the lowest Adj Close is 174.679993<br />
For HD, the lowest Adj Close is 146.575485<br />

1.3

In [84]:
df["LofAC"]=df["Adj Close"].log()

AttributeError: 'Series' object has no attribute 'log'

1.4

In [79]:
df.groupby("Company").idxmax()

High         Low        Open       Close      Volume  \
Company                                                               
AME      2021-12-29  2021-12-30  2021-12-30  2021-12-29  2020-06-26   
APH      2021-12-30  2021-12-31  2021-12-30  2021-12-29  2017-11-30   
ELV      2022-10-31  2022-10-31  2022-11-01  2022-10-31  2019-04-12   
HD       2021-12-06  2021-12-07  2021-12-07  2021-12-07  2022-02-22   

          Adj Close  
Company              
AME      2021-12-29  
APH      2021-12-29  
ELV      2022-10-31  
HD       2021-12-07

For AME, the lowest Adj Close is 57.247528 <br />
For APH, the lowest Adj Close is 33.339298 <br />
For ELV, the lowest Adj Close is 174.679993<br />
For HD, the lowest Adj Close is 146.575485<br />

In [80]:
df.groupby("Company").idxmin()

High         Low        Open       Close      Volume  \
Company                                                               
AME      2020-03-23  2020-03-18  2020-03-18  2020-03-23  2019-12-24   
APH      2020-03-18  2020-03-18  2020-03-18  2020-03-16  2019-12-24   
ELV      2020-03-23  2020-03-23  2020-03-24  2020-03-23  2020-12-24   
HD       2018-12-24  2020-03-18  2020-03-23  2020-03-20  2020-12-24   

          Adj Close  
Company              
AME      2020-03-23  
APH      2020-03-16  
ELV      2020-03-23  
HD       2020-03-20